app/assets/javascripts/discourse/app/lib/text.js

```diff
@@ -110,3 +110,55 @@ export function emojiUrlFor(code) {
     return buildEmojiUrl(code, opts);
   }
 }
+
+function encode(str) {
+  return str.replaceAll("<", "&lt;").replaceAll(">", "&gt;");
+}
+
+function traverse(element, callback) {
+  if (callback(element)) {
+    for (let i = 0; i < element.childNodes.length; ++i) {
+      traverse(element.childNodes[i], callback);
+    }
+  }
```

Comment:
Any reasons not to use ` forEach`  instead?

` ` ` 
element.childNodes.forEach(child => traverse(child, callback))
` ` `